src/pages/Admin/Admin.tsx

```diff
@@ -409,26 +423,28 @@ const VenueInfoComponent: React.FC<AdminVenueDetailsPartProps> = ({
 
 const Admin: React.FC = () => {
   const { user } = useUser();
+  const userId = user?.uid || "";
+
+  const { isAdminUser, isLoading: isAdminUserLoading } = useIsAdminUser(userId);
```

Comment:
This replaces `useRoles` from the old code, in hopefully a simpler to understand/more performant way.

Reply:
Even if we need more in future, these similar patterns may work 'better' for our usages/etc. At some point we may want to deprecate/remove/cleanup `useRole` in favour of this (I know it was used elsewhere in the code, though was out of scope (OOS) for this PR)